orangecontrib/text/widgets/owcorpusviewer.py

```diff
@@ -304,7 +304,8 @@ def update_info(self):
     def commit(self):
         if self.corpus is not None:
             matched = self.corpus[self.output_mask]
-            unmatched_mask = [i for i in range(len(self.corpus)) if i not in self.output_mask]
+            output_mask = set(self.output_mask)
+            unmatched_mask = [i for i in range(len(self.corpus)) if i not in output_mask]
```

Comment:
If `unmatched_mask` could be a numpy array of booleans (which Corpus (Table) currently doesn't support, but numpy/pandas would), this could be even faster:
```py
unmatched_mask = np.ones(len(self.corpus), dtype=bool)
unmatched_mask[self.output_mask] = 0
```
But I don't suppose this is the bottleneck anymore. :smile: